dlib/serialize.h

```diff
@@ -1772,8 +1889,321 @@ namespace dlib
+            int version_int = 0;                                        \
+            deserialize(version_int, in);                               \
+            if (version_int != version)                                 \
+                throw dlib::serialization_error("Unexpected version found while deserializing " #Type); \
+            DLIB_EXPAND(DLIB_PASTE(DLIB_FROM, __VA_ARGS__))             \
+        }                                                               \
+        catch (dlib::serialization_error& e)                            \
+        {                                                               \
+            throw dlib::serialization_error(e.info + "\n   while deserializing object of type " #Type); \
+        }                                                               \
+    }
+
+#elif DLIB_DEFAULT_SERIALIZATION_METHOD == 1
+
+#define DLIB_DEFINE_DEFAULT_SERIALIZATION(Type, version, ...)           \
+    template<typename T>                                                \
+    void serialize_me(std::ostream& out, const T& x) const              \
+    {                                                                   \
+        using dlib::serialize;                                          \
+        serialize(x, out);                                              \
+    }                                                                   \
+                                                                        \
+    template<typename T, typename... Rest>                              \
+    void serialize_me(std::ostream& out, const T& x, const Rest& ... rest) const  \
+    {                                                                   \
+        serialize_me(out, x);                                           \
+        serialize_me(out, rest...);                                     \
+    }                                                                   \
+                                                                        \
+    template<typename T>                                                \
+    void deserialize_me(std::istream& in, T& x)                         \
+    {                                                                   \
+        using dlib::deserialize;                                        \
+        deserialize(x, in);                                             \
+    }                                                                   \
+                                                                        \
+    template<typename T, typename... Rest>                              \
+    void deserialize_me(std::istream& in, T& x, Rest& ... rest)         \
+    {                                                                   \
+        deserialize_me(in, x);                                          \
+        deserialize_me(in, rest...);                                    \
+    }                                                                   \
```

Comment:
These methods don't need to be class members.  Just make them global (and rename them to `serialize_these` or some other more appropriate name.  Then this version of DLIB_DEFAULT_SERIALIZATION_METHOD will be even simpler than DLIB_DEFAULT_SERIALIZATION_METHOD 2.